src/services/worker/tasks/liquidation.task.ts

```diff
@@ -1,113 +1,149 @@
 import { LessThanOrEqual } from 'typeorm';
 import { Injectable, Logger } from '@nestjs/common';
-import Web3Provider from '@cli/providers/web3.provider';
 import { Cluster } from '@cli/modules/clusters/cluster.entity';
 import { ConfService } from '@cli/shared/services/conf.service';
-import { RetryService } from '@cli/shared/services/retry.service';
 import { ClusterService } from '@cli/modules/clusters/cluster.service';
 import { MetricsService } from '@cli/modules/webapp/metrics/services/metrics.service';
+import { SystemService, SystemType } from '@cli/modules/system/system.service';
+import { Web3Provider } from '@cli/shared/services/web3.provider';
 
 @Injectable()
 export class LiquidationTask {
+  private static isProcessLocked = false;
   private readonly _logger = new Logger(LiquidationTask.name);
+
   constructor(
     private _config: ConfService,
     private _metrics: MetricsService,
-    private _retryService: RetryService,
     private _clusterService: ClusterService,
+    private _systemService: SystemService,
+    private _web3Provider: Web3Provider,
   ) {}
 
+  static get BLOCK_RANGE() {
+    return 10;
+  }
+
   async liquidate(): Promise<void> {
-    const currentBlockNumber = +(await this._retryService.getWithRetry(
-      Web3Provider.currentBlockNumber,
-    ));
-    const toLiquidateRecords = await this._clusterService.findBy({
-      where: {
-        liquidationBlockNumber: LessThanOrEqual(
-          // Current block
-          // than the block number when balance becomes zero if not liquidated
-          currentBlockNumber,
-        ),
-      },
-    });
-    const clustersToLiquidate: Cluster[] = [];
-    const alreadyLiquidatedClusterUpdates = {
-      balance: null,
-      burnRate: null,
-      isLiquidated: true,
-      liquidationBlockNumber: null,
-    };
-    for (const item of toLiquidateRecords) {
-      item.cluster = JSON.parse(item.cluster);
-      const logItem = JSON.stringify(item);
-      try {
-        this._logger.log(
-          `Checking in a contract that cluster is liquidatable: ${logItem}`,
-        );
-        const canLiquidateCluster = await this._retryService.getWithRetry(
-          Web3Provider.liquidatable,
-          [item.owner, item.operatorIds, item.cluster],
-        );
-        if (canLiquidateCluster) {
-          this._logger.log(`YES. Cluster is liquidatable: ${logItem}`);
-          clustersToLiquidate.push(item);
-        } else {
+    if (LiquidationTask.isProcessLocked) {
```

Comment:
The function is long, it can be refactored, for example can extract fetching latestSyncedBlockNumber and latestBlockNumber and checking if it is in sync to a function.